include/ignition/gazebo/detail/View.hh

```diff
@@ -62,36 +69,45 @@ class View : public BaseView
   /// \param[_in] _entity The entity
   /// \return The entity and its component data. Const pointers to the component
   /// data are returned.
-  public: ConstComponentData EntityComponentConstData(
-              const Entity _entity) const;
+  public: const std::vector<const components::BaseComponent *>
+          &EntityComponentConstData(const Entity _entity) const;
 
   /// \brief Get an entity and its component data. It is assumed that the entity
   /// being requested exists in the view.
   /// \param[_in] _entity The entity
   /// \return The entity and its component data. Mutable pointers to the
   /// component data are returned.
-  public: ComponentData EntityComponentData(const Entity _entity);
+  public: const std::vector<components::BaseComponent *>
```

Comment:
nit: use the `ComponentData` alias?

Also, should the return reference be marked `const`? I assume by marking the return reference `const`, that means pointers can't be added/removed to/from the vector, but the existing pointers in the data can be modified since the pointers are not const, right?

Reply:
Done in https://github.com/ignitionrobotics/ign-gazebo/pull/1317/commits/48b2befd4efaed475c632058ae1783193eecdd55. 

> should the return reference be marked const? 

It's already `const`. Are you saying it shouldn't be?

Reply:
What you're doing here looks correct to me - sorry, I misread this the first time around. I'll mark this as resolved :+1: